controller/instance_manager_controller.go

```diff
@@ -1702,34 +1701,52 @@ func (imc *InstanceManagerController) deleteOrphans(im *longhorn.InstanceManager
 	return nil
 }
 
-func (imc *InstanceManagerController) isOrphanDeletable(orphan *longhorn.Orphan, autoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isDeletable bool, err error) {
-	if autoDelete {
-		return true, nil
+func (imc *InstanceManagerController) deleteOrphan(orphan *longhorn.Orphan, resourceDeletable bool) error {
+	if orphan.Spec.ResourceDeletable != resourceDeletable {
+		orphan.Spec.ResourceDeletable = resourceDeletable
+		_, err := imc.ds.UpdateOrphan(orphan)
+		switch {
+		case datastore.ErrorIsNotFound(err):
+			return nil
+		case err != nil:
+			return errors.Wrapf(err, "failed to update resource deletable on orphan %v before deletion", orphan.Name)
+		}
 	}
 
+	err := imc.ds.DeleteOrphan(orphan.Name)
+	switch {
+	case datastore.ErrorIsNotFound(err):
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to delete orphan %v", orphan.Name)
+	}
+	return nil
+}
```

Comment:
Not sure about deleting the orphan CR blindly. Maybe consider adding a condition such as `resourceDeleted` to orphan.State.Conditions to make the deletion more interpretable.

Reply:
This is what `spec.resourceDeletable` means, and another way is to reuse `DataCleanable` condition on data orphans. But, wouldn't it violate the convention to update status outside of orphan controller?

Reply:
We'll keep blind cleanup with UUID in the next PR, and let instance manager protect the processes instead.